src/librbd/io/ImageRequest.cc

```diff
@@ -240,36 +239,13 @@ void ImageRequest<I>::send() {
     return;
   }
 
-  if (m_bypass_image_cache || m_image_ctx.image_cache == nullptr) {
-    update_timestamp();
-    send_request();
-  } else {
-    send_image_cache_request();
-  }
+  update_timestamp();
+  send_request();
 }
 
 template <typename I>
 int ImageRequest<I>::clip_request() {
-  std::shared_lock image_locker{m_image_ctx.image_lock};
-  for (auto &image_extent : m_image_extents) {
-    auto clip_len = image_extent.second;
-    int r = clip_io(get_image_ctx(&m_image_ctx), image_extent.first, &clip_len);
-    if (r < 0) {
-      return r;
-    }
-
-    image_extent.second = clip_len;
-  }
-  return 0;
-}
-
-template <typename I>
-uint64_t ImageRequest<I>::get_total_length() const {
-  uint64_t total_bytes = 0;
-  for (auto& image_extent : this->m_image_extents) {
-    total_bytes += image_extent.second;
-  }
-  return total_bytes;
+  return util::clip_request(&m_image_ctx, m_image_extents);
```

Comment:
Nit: see previous comment about doing this logic in one place only one time

Reply:
Same as above.